src/CommunityToolkit.Aspire.Hosting.Solr/SolrBuilderExtensions.cs

```diff
@@ -0,0 +1,62 @@
+using Aspire.Hosting.ApplicationModel;
+using CommunityToolkit.Aspire.Hosting.Solr;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extension methods for adding and configuring a Solr resource.
+/// </summary>
+public static class SolrBuilderExtensions
+{
+    /// <summary>
+    /// Adds an Apache Solr container resource to the distributed application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port for Solr.</param>
+    /// <param name="coreName">The name of the core to create.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{SolrResource}"/>.</returns>
+    public static IResourceBuilder<SolrResource> AddSolr(this IDistributedApplicationBuilder builder, [ResourceName] string name, int? port = null, string? coreName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        if (string.IsNullOrEmpty(coreName))
+        {
+            coreName = "solr";
+        }
+
+        var resource = new SolrResource(name, coreName);
+
+        builder.Eventing.Subscribe<ConnectionStringAvailableEvent>(resource, async (_, ct) =>
+        {
+            var connectionString = await resource.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+
+            if (connectionString is null)
+            {
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{resource.Name}' resource but the connection string was null.");
+            }
+        });
```

Comment:
This is just validating some internal logic of Aspire and isn't needed. We have it elsewhere so that we can grab the connection string and push it to a client for health checks.